build/codegen/codegen.sh

```diff
@@ -14,29 +14,34 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-GROUP_VERSIONS="ceph.rook.io:v1"
+set -o errexit
+set -o nounset
+set -o pipefail
 
-scriptdir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+script_dir="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 
-# CODE GENERATION
-# we run deepcopy and client,lister,informer generations separately so we can use the flag "--plural-exceptions"
-# which is only known by client,lister,informer binary and not the deepcopy binary
+base_dir="$(cd "${script_dir}/../.."  && pwd)"
 
+boilerplate="${base_dir}"/build/codegen/header.txt
+
+source "${CODE_GENERATOR}/kube_codegen.sh"
+
+
+(
+  # because apis is a separate go module, code generation
+  # must be run from within the corresponding directory.
+  cd  ${base_dir}/pkg/apis
 # run code deepcopy generation
```

Comment:
The deepcopy comment doesn't relate to the `cd`, so these lines should be separated.

Reply:
done